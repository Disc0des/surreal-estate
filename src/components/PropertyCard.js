import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import "../styles/PropertyCard.css";
import house from "../2-kings.png";

function PropertyCard({
  title,
  type,
  bedrooms,
  bathrooms,
  city,
  price,
  email,
}) {
  const sendEmail = () => {
    window.open(`mailto:${email}`);
  };

  return (
    <div className="property-card">
      <img src={house} alt="house" className="two-kings" />
      <div className="details-cont">
        <ul className="icon-cont">
          <li>
            <FontAwesomeIcon icon="fa-solid fa-tag" />
          </li>
          <li>
            <FontAwesomeIcon icon="fa-solid fa-map-pin" />
          </li>
          <li>
            <FontAwesomeIcon icon="fa-solid fa-house" />
          </li>
          <li>
            <FontAwesomeIcon icon="fa-solid fa-bed" />
          </li>
          <li>
            <FontAwesomeIcon icon="fa-solid fa-bath" />
          </li>
          <li>
            <FontAwesomeIcon icon="fa-solid fa-sterling-sign" />
          </li>
        </ul>
        <ul className="property-details">
          <li className="title-details">{title}</li>
          <li className="city-details">{city}</li>
          <li className="type-details">{type}</li>
          <li className="beds-details">{bedrooms}</li>
          <li className="bath-details">{bathrooms}</li>
          <li className="price-details">{price}</li>
        </ul>
      </div>
      <button className="email-button" type="button" onClick={sendEmail}>
        <FontAwesomeIcon icon="fa-solid fa-envelope" />
      </button>
    </div>
  );
}

export default PropertyCard;

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  bathrooms: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  email: PropTypes.string.isRequired,
};
