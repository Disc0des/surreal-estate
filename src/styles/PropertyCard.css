.property-card {
  background-color: #d2adee;
  width: 200px;
}