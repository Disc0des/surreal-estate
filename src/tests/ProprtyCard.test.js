import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("property card render", () => {
  const validProps = {
    title: "dummy-title",
    type: "dummy-type",
    bedrooms: 3,
    bathrooms: 2,
    price: 100000,
    city: "dummy-city",
    email: "dummy@dummy.com",
  };
  it("renders property card correctly", () => {
    const { asFragment } = render(<PropertyCard {...validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct information for each prop", () => {
    const { getByText } = render(<PropertyCard {...validProps} />);
    expect(getByText("dummy-title")).toHaveClass("card-title");
    expect(getByText("dummy-city")).toHaveClass("card-city");
    expect(getByText("dummy-type")).toHaveClass("card-type");
    expect(getByText(3)).toHaveClass("card-beds");
    expect(getByText(2)).toHaveClass("card-baths");
    expect(getByText(100000)).toHaveClass("card-price");
    expect(getByText("dummy-city")).toHaveClass("card-city");
    expect(getByText("dummy@dummy.com")).toHaveClass("card-email");
  });
});
