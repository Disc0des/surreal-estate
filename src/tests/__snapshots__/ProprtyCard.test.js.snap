// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`property card render renders property card correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <img
      alt="house"
      class="2-kings"
      src="2-kings.png"
    />
    <ul
      class="property-details"
    >
      <li
        class="card-title"
      >
        dummy-title
      </li>
      <li
        class="card-type"
      >
        dummy-type
      </li>
      <li
        class="card-beds"
      >
        3
      </li>
      <li
        class="card-baths"
      >
        2
      </li>
      <li
        class="card-city"
      >
        dummy-city
      </li>
      <li
        class="card-price"
      >
        100000
      </li>
      <li
        class="card-email"
      >
        dummy@dummy.com
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
